File: Greedy/곱하기혹은더하기.py
arr = list(map(int, input()))

def solution(result, arr):
    if len(arr) == 1:
        return arr[0]
    result = max(arr[0] + arr[1], arr[0] * arr[1])
    for i in range(2, len(arr)):
        if result + arr[i] < result * arr[i]:
            result *= arr[i]
        else:
            result += arr[i]
    return result
print(solution(result, arr))

File: Greedy/곱하기혹은더하기.js
let arr = prompt()
  .split("")
  .map((v) => parseInt(v));

function solution(arr) {
  if (arr.length == 1) return arr[0];
  let result = Math.max(arr[0] + arr[1], arr[0] * arr[1]);
  for (let i = 2; i < arr.length; i += 1) {
    if (result + arr[i] < result * arr[i]) {
      result *= arr[i];
    } else {
      result += arr[i];
    }
  }
  return result;
}
